File: 1529_minFlips.java
class Solution {
    public int minFlips(String target) {
        int count = 0;
        char last_bit = target.charAt(0);
        for (int i = 0; i < target.length(); i++) {
            if (last_bit == target.charAt(i)) {
                continue;
            } else {
                count++;
                last_bit = target.charAt(i);
            }
        }
        if (target.charAt(0) == '0') {
            return count;
        } else {
            return count + 1;
        } 
    }
}

File: 1501-2000/1529_minFlips.py
class Solution:
    def minFlips(self, target: str) -> int:
        count = 0
        last_bit = target[0]
        for bit in target:
            if last_bit == bit:
                continue
            else:
                count += 1
                last_bit = bit
        return count if target[0] == "0" else count + 1
